pulp_2to3_migration/app/pre_migration.py

```diff
@@ -185,8 +184,8 @@ def delete_removed_pulp2_content(content_model):
                 pulp2content_pb.total -= 1
                 pulp2detail_pb.total -= 1
                 continue
-
-        downloaded = record.downloaded if hasattr(record, 'downloaded') else False
+        is_downloaded = hasattr(record, 'downloaded') and getattr(record, 'downloaded')
```

Comment:
Content that does not have set `downloaded` flag and does not have lazy_catalog_entry will not be migrated. That's basically all the content prior to https://github.com/pulp/pulp/pull/2147/files#diff-15dec39bbc93d3954990163924a66076cfbcda47abcef37968377c68517f3466R647.

There is another path, which will consist tin checking existence of the storage_path for every content that has `downloaded=null`

Reply:
@jlsherrill can you weigh in here?

Reply:
That is...a really old PR up there. Great find!

Reply:
or, actually we can set downloaded to True for every record that does not have set this field. If the storage_path will turn out to be missing or corrupted user will need to re-sync those super-old repos or just use the skip-corrupted flag

Reply:
yeah, I wanted to say that downloaded is null because we didn't have on_demand notion at all and all content was downloaded. +1 to set it to True.

Reply:
Yeah, if lazy sync was added at the same (in the same release) then we can assume that prior to that everything was downloaded.